sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRel.java

```diff
@@ -210,33 +211,57 @@ public int getCount() {
   }
 
   private class LimitTransform<T> extends PTransform<PCollection<T>, PCollection<T>> {
+    private int startIndex;
+
+    public LimitTransform(int startIndex) {
+      this.startIndex = startIndex;
+    }
+
     @Override
     public PCollection<T> expand(PCollection<T> input) {
       Coder<T> coder = input.getCoder();
       PCollection<KV<String, T>> keyedRow =
           input.apply(WithKeys.of("DummyKey")).setCoder(KvCoder.of(StringUtf8Coder.of(), coder));
 
-      return keyedRow.apply(ParDo.of(new LimitFn<T>(getCount())));
+      return keyedRow.apply(ParDo.of(new LimitFn<T>(getCount(), startIndex)));
     }
   }
 
   private static class LimitFn<T> extends DoFn<KV<String, T>, T> {
     private final Integer limitCount;
+    private final Integer startIndex;
 
-    public LimitFn(int c) {
+    public LimitFn(int c, int s) {
       limitCount = c;
+      startIndex = s;
     }
 
     @StateId("counter")
     private final StateSpec<ValueState<Integer>> counterState = StateSpecs.value(VarIntCoder.of());
 
+    @StateId("skipped_rows")
+    private final StateSpec<ValueState<Integer>> skippedRowsState =
```

Comment:
Is there a way to initialize state directly?

Reply:
I don't think there is. That is a good feature! It would have to be serialized w/ the coder for graph submission. Totally doable. Maybe good starter task.

Reply:
It is also a useful feature.

Reply:
Ah, it is very old and is https://issues.apache.org/jira/browse/BEAM-1335